### ISSUE TITLE
Track title no longer updating when used with players like mpd or mpv

### ISSUE THREAD

**Issue by Earnestly** [rmcauley/rainwave#160]
For some time now the title/artist metadata no longer updates to reflect what is playing, instead the title falls back to the url such as `http://allrelays.rainwave.cc/all.ogg`.

Is this a known issue or something that can be fixed?

**Comment by rmcauley** [comment]
When we upgraded to the Opus codec, we found out pretty quickly that unfortunately Chrome (and a few others - but most importantly Chrome) did not support metadata/Ogg tags as part of the Opus stream to be supported.

Apparently this can be worked around with old, old, old school ICY tags for old, old, old school players like MPD and Foobar.

Here's the Chromium issue: https://bugs.chromium.org/p/chromium/issues/detail?id=1161429&q=opus&can=2

It's one in a long line of duplicates I've seen closed over the last few years. :)

**Comment by rmcauley** [comment]
Looking over that bug report, it seems to have devolved entirely.  I may have to spin up a local Rainwave and provide a much more concrete bug report to the Chromium team.

**Comment by Earnestly** [comment]
Yeah it's nice to have opus I agree.  Am I understandingly correctly that metadata was removed so that chromium itself could handle the stream?

So that if chromium managed to fix this bug then the metadata can eventually be reinstated?

**Comment by rmcauley** [comment]
> Am I understandingly correctly that metadata was removed specifically from opus so that chromium itself could 
handle the stream? And that if chromium managed to fix this bug then the metadata can eventually be reinstated?

Yes, that is correct.

> If this the case then I think it's perfectly reasonable to wait on chromium. It appears the mp3 stream still works (using icy-title tags) so I'll use that in the mean time.

Yes, that is also correct.  You will not have the same audio quality as the Opus stream, but you will have metadata.

**Comment by Earnestly** [comment]
This much is true.  The issue might be worth keeping simply to cross-reference the chromium bug until that is fixed, which is the proper place for it.

PS: Thank you for the detailed response and explaining the reasons for this, it does seem a bit esoteric otherwise.

**Comment by rmcauley** [comment]
More than happy to provide a detailed response. 😊

**Comment by haihige** [comment]
This doesn't seem to be working for some of the stations (All, Chiptune, and Covers), even with the mp3 stream. This is what I see in mpv:

```
Playing: https://relay.rainwave.cc:443/all.mp3
 ● Audio  --aid=1  (mp3 2ch 44100 Hz 116 kbps)
File tags:
 icy-title: Unknown
```

**Comment by rmcauley** [comment]
Hahah pasted the password into the comment, good job Rob :/

**Comment by Earnestly** [comment]
The information is retained in the comment revision history; see https://docs.github.com/en/communities/moderating-comments-and-conversations/tracking-changes-in-a-comment

**Comment by rmcauley** [comment]
Thanks for the heads up.  I changed the server password and restarted the streams with the new password. :)
